services/kuber/server/server.go

```diff
@@ -55,96 +61,121 @@ func (s *server) SetUpStorage(ctx context.Context, req *pb.SetUpStorageRequest)
 	}
 	if err := errGroup.Wait(); err != nil {
 		log.Error().Msgf("Error encountered in SetUpStorage : %s", err.Error())
-		return &pb.SetUpStorageResponse{DesiredState: desiredState, ErrorMessage: fmt.Sprintf("Error encountered in SetUpStorage: %s", err.Error())}, err
+		return &pb.SetUpStorageResponse{DesiredState: desiredState}, err
 	}
 	log.Info().Msgf("Storage was successfully set up for project %s", desiredState.Name)
-	return &pb.SetUpStorageResponse{DesiredState: desiredState, ErrorMessage: ""}, nil
+	return &pb.SetUpStorageResponse{DesiredState: desiredState}, nil
+}
+
+func (s *server) StoreClusterMetadata(ctx context.Context, req *pb.StoreClusterMetadataRequest) (*pb.StoreClusterMetadataResponse, error) {
+	if namespace := envs.Namespace; namespace == "" {
+		return &pb.StoreClusterMetadataResponse{}, nil
+	}
+
+	md := ClusterMetadata{
+		NodeIps:    make(map[string][]net.IP),
+		PrivateKey: req.GetCluster().GetClusterInfo().GetPrivateKey(),
+	}
+
+	for _, pool := range req.GetCluster().GetClusterInfo().GetNodePools() {
+		for _, node := range pool.GetNodes() {
+			md.NodeIps[pool.Name] = append(md.NodeIps[pool.Name], net.ParseIP(node.Public))
+		}
+	}
+
+	b, err := json.Marshal(md)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal cluster metadata: %w", err)
```

Comment:
An error can be a little more specific, i.e. `fmt.Errorf("failed to marshal %s cluster metadata: %w", req.GetCluster().GetClusterInfo().GetName(), err)`